[Bug] Event List information Query data exception

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.


### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

The agent collects data and the server cannot serialize it with json

### What you expected to happen

event list Normal query



### How to reproduce

Using JVM information:
IBM J9 VM (build 2.9, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20211022_15212 (JIT enabled, AOT enabled) IBM J9 VM (Build 2.9, JRE 1.8.0 Linux AMD64-64-bit Compressed References 20211022_15212 (JIT Enabled, aOT Enabled)

```
2023-01-11 09:02:30,628 notprivacysafe.graphql.execution.SimpleDataFetcherExceptionHandler 27 [armeria-eventloop-epoll-4-52] WARN [] - Exception while fetching data (/data) : com.google.gson.stream.MalformedJsonException: Unterminated string at line 1 column 2001 path $.
com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unterminated string at line 1 column 2001 path $.
at com.google.gson.Gson.fromJson(Gson.java:1006) ~[gson-2.9.0.jar:?]
at com.google.gson.Gson.fromJson(Gson.java:956) ~[gson-2.9.0.jar:?]
at com.google.gson.Gson.fromJson(Gson.java:905) ~[gson-2.9.0.jar:?]
at org.apache.skywalking.oap.server.core.query.type.event.Event.setParameters(Event.java:58) ~[server-core-9.2.0.jar:9.2.0]
at org.apache.skywalking.oap.server.storage.plugin.elasticsearch.query.ESEventQueryDAO.parseSearchHit(ESEventQueryDAO.java:179) ~[storage-elasticsearch-plugin-9.2.0.jar:9.2.0]
at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:?]
at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source) ~[?:?]
at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:?]
at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:?]
at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:?]
at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:?]
at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:?]
at org.apache.skywalking.oap.server.storage.plugin.elasticsearch.query.ESEventQueryDAO.getEventsResultByCurrentBuilder(ESEventQueryDAO.java:75) ~[storage-elasticsearch-plugin-9.2.0.jar:9.2.0]
at org.apache.skywalking.oap.server.storage.plugin.elasticsearch.query.ESEventQueryDAO.queryEvents(ESEventQueryDAO.java:58) ~[storage-elasticsearch-plugin-9.2.0.jar:9.2.0]
at org.apache.skywalking.oap.server.core.query.EventQueryService.queryEvents(EventQueryService.java:56) ~[server-core-9.2.0.jar:9.2.0]
at org.apache.skywalking.oap.query.graphql.resolver.EventQuery.queryEvents(EventQuery.java:49) ~[query-graphql-plugin-9.2.0.jar:9.2.0]
at jdk.internal.reflect.GeneratedMethodAccessor167.invoke(Unknown Source) ~[?:?]
at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
at graphql.kickstart.tools.resolver.MethodFieldResolverDataFetcher.get(MethodFieldResolver.kt:280) ~[graphql-java-tools-12.0.2.jar:?]
at graphql.execution.ExecutionStrategy.fetchField(ExecutionStrategy.java:279) ~[graphql-java-17.3.jar:?]
at graphql.execution.ExecutionStrategy.resolveFieldWithInfo(ExecutionStrategy.java:210) ~[graphql-java-17.3.jar:?]
at graphql.execution.AsyncExecutionStrategy.execute(AsyncExecutionStrategy.java:60) ~[graphql-java-17.3.jar:?]
at graphql.execution.Execution.executeOperation(Execution.java:159) ~[graphql-java-17.3.jar:?]
at graphql.execution.Execution.execute(Execution.java:105) ~[graphql-java-17.3.jar:?]
at graphql.GraphQL.execute(GraphQL.java:613) ~[graphql-java-17.3.jar:?]
at graphql.GraphQL.parseValidateAndExecute(GraphQL.java:538) ~[graphql-java-17.3.jar:?]
at graphql.GraphQL.executeAsync(GraphQL.java:502) ~[graphql-java-17.3.jar:?]
at com.linecorp.armeria.server.graphql.DefaultGraphqlService.execute(DefaultGraphqlService.java:101) ~[armeria-graphql-1.16.0.jar:?]
at com.linecorp.armeria.server.graphql.DefaultGraphqlService.executeGraphql(DefaultGraphqlService.java:93) ~[armeria-graphql-1.16.0.jar:?]
at com.linecorp.armeria.server.graphql.protocol.AbstractGraphqlService.lambda$doPost$0(AbstractGraphqlService.java:117) ~[armeria-graphql-protocol-1.16.0.jar:?]
at java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
at com.linecorp.armeria.common.util.UnmodifiableFuture.doComplete(UnmodifiableFuture.java:152) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.util.UnmodifiableFuture.lambda$wrap$0(UnmodifiableFuture.java:103) ~[armeria-1.16.0.jar:?]
at java.util.concurrent.CompletableFuture.uniHandle(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture$UniHandle.tryFire(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
at com.linecorp.armeria.common.stream.StreamMessageCollector.onComplete(StreamMessageCollector.java:67) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.AbstractStreamMessage$CloseEvent.notifySubscriber(AbstractStreamMessage.java:258) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberOfCloseEvent0(DefaultStreamMessage.java:300) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriberOfCloseEvent(DefaultStreamMessage.java:292) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.handleCloseEvent(DefaultStreamMessage.java:429) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.notifySubscriber0(DefaultStreamMessage.java:372) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.doRequest(DefaultStreamMessage.java:275) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.request(DefaultStreamMessage.java:257) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.AbstractStreamMessage$SubscriptionImpl.request(AbstractStreamMessage.java:211) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.StreamMessageCollector.onSubscribe(StreamMessageCollector.java:52) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.subscribe(DefaultStreamMessage.java:149) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.stream.DefaultStreamMessage.lambda$subscribe$0(DefaultStreamMessage.java:135) ~[armeria-1.16.0.jar:?]
at com.linecorp.armeria.common.RequestContext.lambda$makeContextAware$3(RequestContext.java:547) ~[armeria-1.16.0.jar:?]
at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) [netty-common-4.1.77.Final.jar:4.1.77.Final]
at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) [netty-common-4.1.77.Final.jar:4.1.77.Final]
at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) [netty-common-4.1.77.Final.jar:4.1.77.Final]
at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:391) [netty-transport-classes-epoll-4.1.77.Final.jar:4.1.77.Final]
at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995) [netty-common-4.1.77.Final.jar:4.1.77.Final]
at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.77.Final.jar:4.1.77.Final]
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.77.Final.jar:4.1.77.Final]
at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: com.google.gson.stream.MalformedJsonException: Unterminated string at line 1 column 2001 path $.
at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1597) ~[gson-2.9.0.jar:?]
at com.google.gson.stream.JsonReader.nextQuotedValue(JsonReader.java:1031) ~[gson-2.9.0.jar:?]
at com.google.gson.stream.JsonReader.nextString(JsonReader.java:815) ~[gson-2.9.0.jar:?]
at com.google.gson.internal.bind.TypeAdapters$15.read(TypeAdapters.java:393) ~[gson-2.9.0.jar:?]
at com.google.gson.internal.bind.TypeAdapters$15.read(TypeAdapters.java:381) ~[gson-2.9.0.jar:?]
at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[gson-2.9.0.jar:?]
at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:187) ~[gson-2.9.0.jar:?]
at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:145) ~[gson-2.9.0.jar:?]
at com.google.gson.Gson.fromJson(Gson.java:991) ~[gson-2.9.0.jar:?]
... 60 more
```

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/docs/en/changes/changes.md
+++ b/docs/en/changes/changes.md
@@ -33,6 +33,8 @@
 * Fix wrong environment variable name `SW_OTEL_RECEIVER_ENABLED_OTEL_RULES` to right `SW_OTEL_RECEIVER_ENABLED_OTEL_METRICS_RULES`.
 * Fix instance query in JDBC implementation.
 * Set the `SW_QUERY_MAX_QUERY_COMPLEXITY` default value to 3000(was 1000).
+* Accept `length=4000` parameter value of the event. It was 2000. 
+* Tolerate parameter value in illegal JSON format.
 
 #### UI
 

--- a/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/Event.java
+++ b/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/metrics/Event.java
@@ -74,7 +74,7 @@ public class Event extends Metrics {
 
     public static final String LAYER = "layer";
 
-    private static final int PARAMETER_MAX_LENGTH = 2000;
+    private static final int PARAMETER_MAX_LENGTH = 4000;
 
     @Override
     protected StorageID id0() {

--- a/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/event/Event.java
+++ b/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/event/Event.java
@@ -19,13 +19,15 @@
 package org.apache.skywalking.oap.server.core.query.type.event;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
 import com.google.gson.reflect.TypeToken;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 import lombok.Data;
-import org.apache.skywalking.oap.server.library.util.StringUtil;
 import org.apache.skywalking.oap.server.core.query.type.KeyValue;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
 
 @Data
 public class Event {
@@ -55,9 +57,18 @@ public class Event {
 
     public void setParameters(final String json) {
         if (StringUtil.isNotEmpty(json)) {
-            final Map<String, String> map = GSON.fromJson(json, new TypeToken<Map<String, String>>() {
-            }.getType());
-            this.parameters = map.entrySet().stream().map(e -> new KeyValue(e.getKey(), e.getValue())).collect(Collectors.toList());
+            try {
+                final Map<String, String> map
+                    = GSON.fromJson(json, new TypeToken<Map<String, String>>() {}.getType());
+                this.parameters = map.entrySet()
+                                     .stream()
+                                     .map(e -> new KeyValue(e.getKey(), e.getValue()))
+                                     .collect(Collectors.toList());
+            } catch (JsonSyntaxException e) {
+                this.parameters = new ArrayList<>(2);
+                this.parameters.add(new KeyValue("json_parse", "false"));
+                this.parameters.add(new KeyValue("raw_parameters", json));
+            }
         }
     }
 }

--- a/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/event/Event.java
+++ b/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/event/Event.java
@@ -58,8 +58,8 @@ public class Event {
     public void setParameters(final String json) {
         if (StringUtil.isNotEmpty(json)) {
             try {
-                final Map<String, String> map
-                    = GSON.fromJson(json, new TypeToken<Map<String, String>>() {}.getType());
+                final Map<String, String> map = GSON.fromJson(json, new TypeToken<Map<String, String>>() {
+                }.getType());
                 this.parameters = map.entrySet()
                                      .stream()
                                      .map(e -> new KeyValue(e.getKey(), e.getValue()))

